packages/horizon/contracts/staking/HorizonStaking.sol

```diff
@@ -803,7 +850,12 @@ contract HorizonStaking is HorizonStakingBase, IHorizonStakingMain {
         uint256 _nThawRequests
     ) private {
         DelegationPoolInternal storage pool = _getDelegationPool(_serviceProvider, _verifier);
-        require(pool.tokens != 0, HorizonStakingInvalidDelegationPoolState(_serviceProvider, _verifier));
+
+        // An invalid delegation pool has shares but no tokens
+        require(
```

Comment:
Not sure if this would behave well with `pool.shares == 0` and `pool.tokens == 0`, see below we do `pool.tokens = pool.tokens - tokensThawed`. So I think we should always enforce `pool.tokens != 0` here?

Reply:
I guess the condition should be `pool.tokens >= tokensThawed`, no?

Theoretically it should always be true because `pool.tokens >= pool.tokensThawing` and `pool.tokensThawing >= tokensThawed`. The only scenario where it gets tricky is when the thawing pool is reset due to the delegation pool being fully slashed, but in that case the nonce invalidation ensures that `tokensThawed = 0`. 

Do you still think we should add a check here?

Reply:
I see, so this would allow clearing up those thaw requests... that makes sense, maybe worth adding a comment here noting we might have a case where `shares == 0` and that then the tokensThawed should be 0?